controller/destination/egress.go

```diff
@@ -0,0 +1,186 @@
+package destination
```

Comment:
I'd consider naming this file dns_watcher.go since the contents of the file make no reference to the term egress